force-app/main/default/classes/ReviewHandler.cls

```diff
@@ -0,0 +1,111 @@
+public class ReviewHandler {
+    // Prevents reviews for applications with 'Rejected' or 'Pending' status
+    public static void preventReviews(List<Review__c> reviews) {
+        Set<Id> rejectedAppIds = new Set<Id>();
+
+        // Collect the IDs of related applications
+        for (Review__c review : reviews) {
+            if (review.Application__c != null) {
+                rejectedAppIds.add(review.Application__c);
+            }
+        }
+
+        // Query the status of related applications
+        Map<Id, Application__c> appMap = new Map<Id, Application__c>([SELECT Id, Status__c FROM Application__c WHERE Id IN :rejectedAppIds]);
+
+        // Iterate through the reviews to verify the application status
+        for (Review__c review : reviews) {
+            if (review.Application__c != null) {
```

Comment:
Do we need to check this , as we have done in line no 8

Reply:
Yes sir , we need to check this otherwise key will be NULL inside appMap.get(key).